crates/biome_css_parser/src/lexer/mod.rs

```diff
@@ -259,6 +267,27 @@ impl<'src> CssLexer<'src> {
         }
     }
 
+    /// Check if the source starts with a Unicode BOM character. If it does,
+    /// consume it and return the UNICODE_BOM token kind.
+    ///
+    /// ## Safety
+    /// Must be called at a valid UT8 char boundary (and realistically only at
+    /// the start position of the source).
+    fn consume_potential_bom(&mut self) -> Option<CssSyntaxKind> {
+        if let Some(first) = self.source().get(0..3) {
```

Comment:
Can we add a comment that tells us why we require inspecting a slice from `0` to `3`? I'm sure there's a reason!